apps/server/src/modules/task/repo/submission.repo.integration.spec.ts

```diff
@@ -1,8 +1,8 @@
 import { EntityManager, ObjectId } from '@mikro-orm/mongodb';
 import { Test, TestingModule } from '@nestjs/testing';
 
+import { Course } from '@src/entities';
```

Comment:
That is because currently the CourseGroup constructor requires a relation to a Course entity:
https://github.com/hpi-schul-cloud/schulcloud-server/blob/9f8820e5dd976a5522e57a54e46c3cf8cc6f5dc7/apps/server/src/modules/task/entity/course-group-info.entity.ts#L10
and the tests apperently require a CourseGroup reference set on the submission:
https://github.com/hpi-schul-cloud/schulcloud-server/blob/bc16c93276a9fbdf114d756a972773e6e863299c/apps/server/src/modules/task/repo/submission.repo.integration.spec.ts#L97
Interestingly the previous code for CourseGroup creation imports the Course entity as well:
https://github.com/hpi-schul-cloud/schulcloud-server/blob/85ab347be31ab21c4b505f12c75e0c55e4d82dab/apps/server/src/entities/learnroom/testHelper.ts#L4
This was just hidden in another class or file.

Reply:
Yes and if it is change you must only change one place and not each. "Hidden" is what is mean with "high quality code". You can use it and must not know anything of the stack with any data flow. This reduce complexity of code and logic. 

If you add one any place all data flows you need a lot of time to implement it. It is impossible to change it. Thats why abstraction is one of the importants point for any code in any situation. If you call it hide, then it is fine. But then please hide every thing that is possible.

Reply:
why shouldn't a test that uses an entiy type not import that type? i really don't see any point in that, sorry.